asf_tools/composite.py

```diff
@@ -0,0 +1,306 @@
+       Use either the given resolution or the largest resolution in the stack"""
+
+    if len(files) < 2:
+        return None
+
+    # Set the pixel size
+    if resolution:
+        pix_size = resolution
+        logging.info(f"Changing pixel size to {pix_size}")
+    else:
+        pix_size = get_max_pixel_size(files)
+        logging.info(f"Using maximum pixel size {pix_size}")
+
+    # Get the median UTM zone and hemisphere
+    home_zone = np.median(parse_zones(files))
+    logging.info(f"Home zone is {home_zone}")
+    hemi = get_hemisphere(files[0])
+    logging.info(f"Hemisphere is {hemi}")
+
+    # Reproject files as needed
+    logging.info("Checking projections")
+    new_files = []
+    for fi in files:
+        fi_name = os.path.split(fi)[1]
+        my_zone = get_zone_from_proj(fi)
+        name = fi_name.replace(".tif", "_reproj.tif")
+        afi = fi.replace(f"_{pol}.tif", "_area.tif")
+        aname = fi_name.replace(f"_{pol}.tif", "_area_reproj.tif")
+        if not os.path.isfile(name):
+            x, y, trans, proj = saa.read_gdal_file_geo(saa.open_gdal_file(fi))
+            if my_zone != home_zone:
+                logging.info(f"Reprojecting {fi} to {name}")
+                if hemi == "N":
+                    proj = ('EPSG:326%02d' % int(home_zone))
+                else:
+                    proj = ('EPSG:327%02d' % int(home_zone))
+                gdal.Warp(name, fi, dstSRS=proj, xRes=pix_size, yRes=pix_size, targetAlignedPixels=True)
+                gdal.Warp(aname, afi, dstSRS=proj, xRes=pix_size, yRes=pix_size, targetAlignedPixels=True)
+                new_files.append(name)
+            elif x < pix_size:
+                # Need to reproject to desired resolution
+                logging.info(f"Changing resolution of {fi} to {pix_size}")
+                gdal.Warp(name, fi, xRes=pix_size, yRes=pix_size, targetAlignedPixels=True)
+                gdal.Warp(aname, afi, xRes=pix_size, yRes=pix_size, targetAlignedPixels=True)
+                new_files.append(name)
+            else:
+                logging.info(f"No reprojection needed; Linking {fi} to {name}")
+                os.symlink(fi, name)
+                os.symlink(afi, aname)
+                new_files.append(name)
+        else:
+            logging.info(f"Found previous reproj file {fi} - taking no action")
+            new_files.append(fi)
+
+    logging.info("All files completed")
+    return new_files
+
+
+def make_composite(outfile, infiles=None, path=None, pol=None, resolution=None):
+
+    '''Create a composite mosaic of infiles using inverse area weighting to adjust backscatter'''
+
+    logging.info(f"make_composite: {outfile} {infiles} {path} {pol} {resolution}")
+    if pol is None:
+        pol = "VV"
+
+    # Establish input file list
+    if path:
+        logging.info("Searching for list of files to process")
+
+        # New format directory names
+        infiles_new = glob.glob(os.path.join(path, f"20*/S1?_IW_*RTC*/*{pol}.tif"))
+
```

Comment:
Should update this glob
* `20*/` is probably not a good assumption
* we could just look for `**/*{pol}.tif` and not worry about directory names